Android/app/src/main/java/com/example/boardgamesocial/DataViews/Adapters/DataClsAdapter.java

```diff
@@ -72,12 +72,27 @@ public void onBindViewHolder(VH holder, int position) {
     }
 
     public void addNewObjects(List<DC> newObjects) {
-        for (DC object : newObjects) {
+        objectList = new ArrayList<>();
+        /** Note: Commented below lines for add functionality
+         * Inorder to have newly added object at the top of the list.
+         * which also helps for delete functionality.
+         * This will help for all CRUD operations on objects.
+         * */
+        /*for (DC object : newObjects) {
             if (!objectList.contains(object)){
                 objectList.add(object);
                 notifyItemChanged(objectList.indexOf(object));
             }
-        }
+
+            objectList.removeIf(obj -> {
+                boolean exists = newObjects.contains(obj);
+                if(!exists)
+                    notifyDataSetChanged();
+                return !exists;
+            });
+        }*/
```

Comment:
I think for removal we should use something like this 
```java
mDataset.remove(position);
notifyItemRemoved(position);
notifyItemRangeChanged(position, mDataSet.size());
```
Here is a link to [StackOverflow](https://stackoverflow.com/questions/26076965/android-recyclerview-addition-removal-of-items)